src/pint/observatory/topo_obs.py

```diff
@@ -197,22 +218,97 @@ def __init__(
         if clock_fmt == "tempo" and clock_file == "time.dat" and tempo_code is None:
             raise ValueError("No tempo_code set for observatory '%s'" % name)
 
+        # GPS corrections
+        self.include_gps = include_gps
+
+        # BIPM corrections
+        # WARNING: `get_observatory` changes these after construction
+        self.include_bipm = include_bipm
+        self.bipm_version = bipm_version
         self.bogus_last_correction = bogus_last_correction
 
         self.tempo_code = tempo_code
+        self.itoa_code = itoa_code
+        if aliases is None:
+            aliases = []
+        for code in (tempo_code, itoa_code):
+            if code is not None:
+                aliases.append(code)
+
         self.origin = origin
+        super().__init__(
+            name,
+            aliases=aliases,
+            include_gps=include_gps,
+            include_bipm=include_bipm,
+            bipm_version=bipm_version,
+            overwrite=overwrite,
+        )
 
     def __repr__(self):
         aliases = [f"'{x}'" for x in self.aliases]
-        s = f"TopoObs('{self.name}' ({','.join(aliases)}) at [{self._loc_itrf.x}, {self._loc_itrf.y} {self._loc_itrf.z}]:\n{self.origin})"
+        s = f"TopoObs('{self.name}' ({','.join(aliases)}) at [{self.location.x}, {self.location.y} {self.location.z}]:\n{self.origin})"
         return s
 
     @property
     def timescale(self):
         return "utc"
 
+    @property
+    def as_dict(self):
```

Comment:
Should these `as_...` be properties or functions? Is this meant to be consistent with Astropy?

Reply:
No idea.

Reply:
I think unless there's a solid reason for them to be properties, these should be functions - they construct a transformed version of the whole object, so it's a little weird that they look like attributes of it.